### PR TITLE
AC-1974 - Global tracking for profile and recipientLists forms

### PR DESCRIPTION
**Pull request by nstanard**
### What Changed

- Added form ids to allow list for all forms in: `2web2ui/src/pages/profile/*` and `2web2ui/src/pages/recipientLists/*`

### How To Test

- Verify that the new forms track in segment when submitted. 
- Verify that the form fields have the name and id attributes along with associated labels where needed. 

### To Do

- [x] Add test for reset 2fa tracking
- [x] Add test for disable 2fa tracking
- [x] Add test for generate 2fa backup codes
- [x] Add test for Edit Profile tracking
- [x] Add test for Password update tracking
- [ ] Feedback

### CODE REVIEW

File: src/components/tracking/constants.js [hunk 1]
```diff
@@ -59,4 +59,10 @@ export const FORM_IDS_ALLOWLIST = [
   'rv-uploadlist-form',
   'register-user-form',
   'events-advanced-filter-form',
+  'profile-update-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:62]: `profile-update-form`
image

File: src/components/tracking/constants.js [hunk 2]
```diff
@@ -59,4 +59,10 @@ export const FORM_IDS_ALLOWLIST = [
...
+  'update-password-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:63]: `update-password-form`
image

Comment (nstanard) [reply src/components/tracking/constants.js:63]: Password form doesn't clear/reset after successfully submitting - we should probably write that up.

File: src/components/tracking/constants.js [hunk 3]
```diff
@@ -59,4 +59,10 @@ export const FORM_IDS_ALLOWLIST = [
...
+  'generate-2fa-backup-codes',
+  'disable-tfa-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:65]: `disable-tfa-form`
image

File: src/components/tracking/constants.js [hunk 4]
```diff
@@ -59,4 +73,11 @@ export const FORM_IDS_ALLOWLIST = [
...
+  'enable-tfa-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:80]: `enable-tfa-form`
image

File: src/components/tracking/constants.js [hunk 5]
```diff
@@ -59,4 +73,11 @@ export const FORM_IDS_ALLOWLIST = [
   'rv-uploadlist-form',
   'register-user-form',
   'events-advanced-filter-form',
+  'profile-update-form',
+  'update-password-form',
+  'generate-2fa-backup-codes',
```
Comment (nstanard) [inline src/components/tracking/constants.js:78]: `generate-2fa-backup-codes`
image

File: src/components/tracking/constants.js [hunk 6]
```diff
@@ -59,4 +73,11 @@ export const FORM_IDS_ALLOWLIST = [
...
+  'disable-tfa-form',
+  'enable-tfa-form',
+  'create-recipientlists-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:81]: `create-recipientlists-form`
image

File: src/components/tracking/constants.js [hunk 7]
```diff
@@ -59,4 +73,11 @@ export const FORM_IDS_ALLOWLIST = [
...
+  'update-recipientlists-form',
```
Comment (nstanard) [inline src/components/tracking/constants.js:82]: `update-recipientlists-form`
image

File: src/components/tracking/constants.js (hunk 4 above)
Comment (nstanard) [reply src/components/tracking/constants.js:77]: We have a possible bug with enabling 2fa: 

I've turned it on and off a number of times now - and at this point if I go to enable it I get an error from the API. But when I refresh the page - it's 100% on.
image
image

**Review by shrirupa — APPROVED** [review]
looks good to me! we need to rebase it, before merging

File: src/components/tracking/constants.js [hunk 8]
```diff
@@ -12,31 +12,36 @@ export const FIELD_IDENTIFIER_BLOCKLIST = [
   'email-address',
   'emailAddress',
   'password',
+  'Password',
```
Comment (nstanard) [inline src/components/tracking/constants.js:15]: I ran the list through a for loop to get rid of dups: 

```

// FIELD_IDENTIFIER_BLOCKLIST
let uniqueList = [];
let dups = [];

for(let i = 0; i < FIELD_IDENTIFIER_BLOCKLIST.length; i++) {
    if(uniqueList.indexOf(FIELD_IDENTIFIER_BLOCKLIST[i]) === -1) {
        uniqueList.push(FIELD_IDENTIFIER_BLOCKLIST[i]);
    } else {
      dups.push(FIELD_IDENTIFIER_BLOCKLIST[i]);
    }
}

```

Comment (shrirupa) [reply src/components/tracking/constants.js:15]: I have verified the fields that are removed

**Review by shrirupa — APPROVED** [review]
looks good!

**Pull request merged**